phoenix-core/src/main/java/org/apache/phoenix/mapreduce/PhoenixRecordWriter.java

```diff
@@ -52,10 +52,20 @@ public PhoenixRecordWriter(final Configuration configuration) throws SQLExceptio
     }
     
     public PhoenixRecordWriter(final Configuration configuration, Set<String> propsToIgnore) throws SQLException {
-        this.conn = ConnectionUtil.getOutputConnectionWithoutTheseProps(configuration, propsToIgnore);
-        this.batchSize = PhoenixConfigurationUtil.getBatchSize(configuration);
-        final String upsertQuery = PhoenixConfigurationUtil.getUpsertStatement(configuration);
-        this.statement = this.conn.prepareStatement(upsertQuery);
+        Connection connection = null;
+        try {
+            connection = ConnectionUtil.getOutputConnectionWithoutTheseProps(configuration, propsToIgnore);
+            this.batchSize = PhoenixConfigurationUtil.getBatchSize(configuration);
+            final String upsertQuery = PhoenixConfigurationUtil.getUpsertStatement(configuration);
+            this.statement = connection.prepareStatement(upsertQuery);
```

Comment:
No exception can be thrown after instantiating the statement object, so that shouldn't be necessary. Besides, we close both the statement and the connection in the `close` method